sdk/core/core-http/lib/credentials/tokenCredential.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { RequestOptionsBase } from "../webResource";
+
+/**
+ * Represents a credential capable of providing an authentication token.
+ */
+export interface TokenCredential {
+  /**
+   * Gets the token provided by this credential.
+   *
+   * @param scopes The list of scopes for which the token will have access.
+   * @param requestOptions The RequestOptionsBase used to configure any requests
+   *                       this TokenCredential implementation might make.
+   */
+  getToken(
+    scopes: string | string[],
+    requestOptions?: RequestOptionsBase
```

Comment:
@schaabs Non http based client libraries will now have to take a dependency on `core-http` to get this interface?

Reply:
cc @bterlson

Reply:
We should probably re-export this interface from `@azure/identity`, developers shouldn't have to depend on `@azure/core-http`.  I'd certainly be in favor of creating a type that's more directly scoped to `identity` if folks are interested.

Reply:
Can you please log an issue to discuss this and get to a conclusion?

Reply:
Actually, now that I think about it, I think the answer for now is to just re-export this type from `@azure/identity`.  Even if we make a more narrowly-scoped type for `getToken` to use, it'd still have to live in `@azure/core-http` to be used in the `TokenCredential` interface.  `RequestOptionsBase` is pretty minimal and only refers to one other type (`TransferProgressEvent`), so it doesn't bring along any major baggage. The properties that it does contain are fairly useful for the purpose of augmenting HTTP requests that are sent through the pipeline.

I've filed issue #3535 to continue this discussion.